ament_cmake_clang_tidy/ament_cmake_clang_tidy-extras.cmake

```diff
@@ -0,0 +1,22 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# copied from ament_cmake_clang_tidy/ament_cmake_clang_tidy-extras.cmake
+
+find_package(ament_cmake_test QUIET REQUIRED)
+
+include("${ament_cmake_clang_tidy_DIR}/ament_clang_tidy.cmake")
+
+ament_register_extension("ament_lint_auto" "ament_cmake_clang_tidy"
```

Comment:
@JShep1 @clalancette this is unfortunate as it may prevent us from backporting without causing havoc on packages that already rely on `ament_lint_auto` :(

Reply:
Is there a strong reason to have it hooked up? If not, and @dirk-thomas agrees, I'd say we remove it to preclude any issues once the patch release is out.

Reply:
A package must be `find_package()`-ing this in order to trigger the extension being registered. So I don't see how this change would be a problem - even with `ament_lint_auto` since no package depends on this new package yet.

Reply:
> A package must be `find_package()`-ing this in order to trigger the extension being registered.

I see it commented out in `ament_lint_common`, yes. 

> So I don't see how this change would be a problem - even with ament_lint_auto since no package depends on this new package yet.

That's true of our packages. Could it ever be a problem if downstream packages start depending on it? If that's too far fetched, then I withdraw my concerns and suggestion.

Reply:
If packages start depending on it they probably intend to use it?

Reply:
Argh, I wrote half of it. I meant to say that it may be a problem if downstream packages start depending on it while building core packages that depend on `ament_lint_auto` from source.

Reply:
How would that effect downstream packages in any way?

Reply:
> A package must be find_package()-ing this in order to trigger the extension being registered.

If a downstream package depends on this linter, it'll get registered as an `ament_lint_auto` extension. If it gets registered as an `ament_lint_auto` extension, it'll get hooked up as a linter for packages that depend on `ament_lint_auto`. That is the case for core packages as well. Or at least that's my train of thought.

Reply:
No package I am aware of declares a linter package as an exported dependency. They are usually only test dependencies.

Reply:
Alright, not a problem then :)